presto-hive/src/main/java/com/facebook/presto/hive/OptimizedLazyBinaryColumnarSerde.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.hive.serde2.ColumnProjectionUtils;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.hive.serde2.SerDeSpec;
+import org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnarStruct;
+import org.apache.hadoop.hive.serde2.lazy.LazySerDeParameters;
+import org.apache.hadoop.hive.serde2.lazybinary.LazyBinaryFactory;
+import org.apache.hadoop.hive.serde2.lazybinary.LazyBinarySerDe;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructField;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.StringObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.io.Writable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * This is a copy of org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnSerDe that avoids
+ * a call to StringObjectInspector.getPrimitiveJavaObject() in serialize() to check whether
+ * the string is empty (it calls getPrimitiveWritableObject() instead). This improves CPU
+ * efficiency by avoiding turning the underlying bytes into a java String.
+ *
+ *
+ * LazyBinaryColumnarSerDe. This serde combines elements of columnar serde and lazybinary serde
+ * to produce a serde which serializes columns into a BytesRefArrayWritable in a compact binary
+ * format and which is deserialized in a lazy, i.e. on-demand fashion.
+ *
+ */
+@SerDeSpec(schemaProps = {serdeConstants.LIST_COLUMNS, serdeConstants.LIST_COLUMN_TYPES})
+public class OptimizedLazyBinaryColumnarSerde
```

Comment:
Move this to `presto-hive-apache` and put in the `org.apache.hive` package, then we don't need `OptimizedColumnarSerDeBase`